Services/Common/OJS.Workers/OJS.Workers.ExecutionStrategies/Java/JavaProjectTestsExecutionStrategy.cs

```diff
@@ -396,6 +428,23 @@ protected virtual void ExtractUserClassNames(string submissionFilePath)
                     .Select(x => x.Contains(".") ? x.Substring(0, x.LastIndexOf(".", StringComparison.Ordinal)) : x)
                     .Select(x => x.Replace("/", ".")));
 
+        private static string ReadAndValidateLine(StringReader output)
+        {
+            var line = output.ReadLine();
+
+            if (line == null)
+            {
+                throw new InvalidOperationException("Unexpected end of output.");
+            }
```

Comment:
This could be more descriptive. In what cases this line could be null? Is this message seen by the end user? Also could be a valid case when the line should be empty (as now you check not only the errorLine with this code..)

Reply:
Done.